tools/swift-plugin-server/Sources/swift-plugin-server/WasmMessageHandler.swift

```diff
@@ -0,0 +1,114 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SystemPackage
+@_spi(PluginMessage) import SwiftCompilerPluginMessageHandling
+
+/// A `PluginMessageHandler` that intercepts messages intended for Wasm plugins.
+final class WasmInterceptingMessageHandler<Base: PluginMessageHandler>: PluginMessageHandler {
```

Comment:
This is effectively letting us chain message handlers, albeit in a specific way. It's fine as-is, but I could see us refactoring this later.

Reply:
Agreed, I was tempted to generalize it but I realized that might be premature — think it's worth following the rule of three here.